src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -443,24 +380,17 @@ private SNITCPHandle CreateTcpHandle(string fullServerName, long timerExpire, ob
                 }
                 catch (Exception e)
                 {
-                    exception = e;
+                    SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.TCP_PROV, SNICommon.InvalidConnStringError, e);
+                    return null;
                 }
             }
 
             SNITCPHandle sniTcpHandle = null;
```

Comment:
Do we still need this variable? You could just return the result directly below.